src/server.c

```diff
@@ -1163,6 +1163,36 @@ static UDSErr_t Handle_0x85_ControlDTCSetting(UDSServer_t *srv, UDSReq_t *r) {
     return UDS_PositiveResponse;
 }
 
+static UDSErr_t Handle_0x87_LinkControl(UDSServer_t *srv, UDSReq_t *r) {
+    if (r->recv_len < UDS_0X85_REQ_BASE_LEN) {
+        return NegativeResponse(r, UDS_NRC_IncorrectMessageLengthOrInvalidFormat);
+    }
+
+    uint8_t type = r->recv_buf[1] & 0x7F;
+
+    if (type == 0x03 && (r->recv_buf[1] & 0x80) == 0) {
```

Comment:
I think it's a good idea to warn here. 

Based on my reading of the standard (14229-1:2020(E) 10.10.1 Step \# 2), I think we should also check that the request is functional before issuing this warning.

`r->info->A_TA_Type == UDS_A_TA_TYPE_FUNCTIONAL`